godtools/App/Features/AppLanguage/Domain/UseCases/GetConfirmAppLanguageInterfaceStringsUseCase.swift

```diff
@@ -0,0 +1,91 @@
+        self.getInterfaceStringRepositoryInterface = getInterfaceStringRepositoryInterface
+        self.localeLanguageName = localeLanguageName
+    }
+    
+    func getStringsPublisher(for selectedLanguage: AppLanguageCodeDomainModel) -> AnyPublisher<ConfirmAppLanguageInterfaceStringsDomainModel, Never> {
+        
+        return Publishers.CombineLatest4(
+            getMessageInNewlySelectedLanguagePublisher(selectedLanguage: selectedLanguage),
+            getMessageInCurrentLanguagePublisher(selectedLanguage: selectedLanguage),
+            getChangeLanguageButtonTextPublisher(selectedLanguage: selectedLanguage),
+            getNevermindButtonTextPublisher(selectedLanguage: selectedLanguage)
+        )
+        .flatMap { messageInNewlySelectedLanguage, messageInCurrenttLanguage, changeLanguageButtonText, nevermindButtonText in
+            
+            let domainModel = ConfirmAppLanguageInterfaceStringsDomainModel(
+                messageInNewlySelectedLanguage: messageInNewlySelectedLanguage,
+                messageInCurrentLanguage: messageInCurrenttLanguage,
+                changeLanguageButtonText: changeLanguageButtonText,
+                nevermindButtonText: nevermindButtonText
+            )
+            
+            return Just(domainModel)
+        }
+        .eraseToAnyPublisher()
+    }
+    
+    private func getMessageInNewlySelectedLanguagePublisher(selectedLanguage: AppLanguageCodeDomainModel) -> AnyPublisher<NSAttributedString, Never> {
+        
+        return getAttributedMessageStringPublisher(selectedLanguage: selectedLanguage, translatedInLanguage: selectedLanguage)
+    }
+    
+    private func getMessageInCurrentLanguagePublisher(selectedLanguage: AppLanguageCodeDomainModel) -> AnyPublisher<NSAttributedString, Never> {
+        
+        return getCurrentAppLanguageUseCase.getLanguagePublisher()
+            .flatMap { currentAppLanguage in
+                
+                return self.getAttributedMessageStringPublisher(selectedLanguage: selectedLanguage, translatedInLanguage: currentAppLanguage)
+            }
+            .eraseToAnyPublisher()
+    }
+    
+    private func getAttributedMessageStringPublisher(selectedLanguage: AppLanguageCodeDomainModel, translatedInLanguage: AppLanguageCodeDomainModel) -> AnyPublisher<NSAttributedString, Never> {
+        
+        return getInterfaceStringRepositoryInterface.getStringPublisher(languageCode: translatedInLanguage, stringId: "languageSettings.confirmAppLanguage.message")
+            .flatMap { formatString in
+                
+                let languageName = self.localeLanguageName.getDisplayName(forLanguageCode: selectedLanguage, translatedInLanguageCode: translatedInLanguage) ?? ""
+                let languageNameAttributed = NSAttributedString(
+                    string: languageName,
+                    attributes: [NSAttributedString.Key.foregroundColor: ColorPalette.gtBlue.uiColor]
```

Comment:
I'm wondering if this NSAttributedString logic should be moved out into the View or ViewModel layer maybe?  I don't know if the coloring here is domain layer specific since the domain layer really shouldn't know anything about the view.  
Let's leave this as is for now and think about that some more and can come back to this UseCase.

Reply:
Yeah I went back and forth on this... the color has to be applied before the string formatting happens, and the string formatting makes sense to be in the useCase.  But I guess both could happen in the viewModel instead?  Or they take turns and the viewModel applies the color and then passes it back to the useCase to format?  Doesn't seem ideal any way you do it.

Reply:
So the use case would end up passing two separate strings, and then the viewModel would apply the color and combine them into 1?  That should work, and would be nice for the domain model to pass a regular string and not an attributed one.

Reply:
Yep the domain model would contain both the message string and the highlight string and then leave it up to the View to apply the highlighting.
We should be able to write a behavior test as well that expects the highlight string to be the newly selected app language in one test and for the other message the highlight string to be the current app language.